### PR TITLE
Implemented the benchmark to observe the effect of middleware over service-client jobs

### PR DESCRIPTION
**Pull request by CihatAltiparmak**
I couldn't test this scenario with rmw_zenoh but in my measurements, Seems rmw_fastrtps is little bit faster than rmw_cyclonedds. 

Edit: Btw i have realised now that i haven't add documentation.
Edit2: Added documentation

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -28,7 +30,8 @@ ament_target_dependencies(
   "benchmark"
   "dynmsg"
   "nav_msgs"
-  "yaml-cpp")
+  "yaml-cpp"
```
Comment (sjahr) [inline CMakeLists.txt:33]: Nitpick: I don't think you need the `"` here and you can remove it from all CMakeLists.txt files (maybe in a separate small cleanup PR)

File: docs/how_to_run.md [hunk 2]
```diff
@@ -7,3 +7,9 @@ This benchmark measures the elapsed time by which the determined path is sent fo
...
+
+### [Basic Service Client Works Benchmark](scenarios/basic_service_client_benchmark.md)
+
+This benchmark measures the total elapsed time based on the time interval between sending the request by the client to the server and getting the response of server. This benchmark utilizes the [ros2/demos](https://github.com/ros2/demos) packages' [example server](https://github.com/ros2/demos/blob/rolling/demo_nodes_cpp/src/services/add_two_ints_server.cpp).
+
+In this benchmark scenario, the benchmarker node only has client interface. THe necessary server for this client is run in [the launch file of this benchmark scenario](../launch/scenario_basic_service_client_benchmark.launch.py). Client sends a request to server and waits for the response from server. Client sends second request to server once the client receives response of first request from client. This actions are repeated `sending_request_number` times. You can configure this `sending_request_number` parameter in [this scenario's launch file]((../launch/scenario_basic_service_client_benchmark.launch.py)).
```
Comment (sjahr) [inline docs/how_to_run.md]:
```suggestion
In this benchmark scenario, the benchmarker node only has client interface. The necessary server for this client is run in [the launch file of this benchmark scenario](../launch/scenario_basic_service_client_benchmark.launch.py). Client sends a request to server and waits for the response from server. Client sends second request to server once the client receives response of first request from client. This actions are repeated `sending_request_number` times. You can configure this `sending_request_number` parameter in [this scenario's launch file]((../launch/scenario_basic_service_client_benchmark.launch.py)).
```

File: docs/scenarios/basic_service_client_benchmark.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+## How To Run Basic Service Client Benchmark
+
+Firstly, source your ros version. It's suggested to test with rolling version of ROS.
```
Comment (sjahr) [inline docs/scenarios/basic_service_client_benchmark.md]:
```suggestion
Firstly, source your ros version. It's suggested to test with rolling version of ROS 2.
```

File: include/moveit_middleware_benchmark/scenarios/scenario_basic_service_client.hpp [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+namespace moveit
+{
+namespace middleware_benchmark
+{
+
+class ScenarioBasicServiceClient
+{
+public:
+  /** \brief Constructor to send request by client to add_two_ints_server which is one
```
Comment (sjahr) [inline include/moveit_middleware_benchmark/scenarios/scenario_basic_service_client.hpp]: Nitpick: If the constructor doesn't do anything unusual and just instantiates members, you can keep it's documentation short by just using

```
/** \brief Constructor
   * \param[in] ...
```

**Comment by CihatAltiparmak** [comment]
Now, this PR is on the way @sjahr , could you take a look at it when you're available?

File: include/moveit_middleware_benchmark/scenarios/scenario_basic_service_client.hpp [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+   * of the example of ros2/demos package
+   *  \param [in] node The ros node for sending request and wait response from server
+   */
+  ScenarioBasicServiceClient(rclcpp::Node::SharedPtr node);
```
Comment (sjahr) [inline include/moveit_middleware_benchmark/scenarios/scenario_basic_service_client.hpp]: Does this make sense?
```suggestion
  ScenarioBasicServiceClient(const rclcpp::Node::SharedPtr& node);
```

**Review by sjahr — COMMENTED** [review]
Just some minor comments. Tag me or Henning again when this is ready for review

File: CMakeLists.txt (hunk 1 above)
Comment (CihatAltiparmak) [reply CMakeLists.txt:33]: Will be solved in small PR

File: include/moveit_middleware_benchmark/scenarios/scenario_basic_service_client.hpp (hunk 4 above)
Comment (CihatAltiparmak) [reply include/moveit_middleware_benchmark/scenarios/scenario_basic_service_client.hpp]: Done it in this commit ( https://github.com/CihatAltiparmak/moveit_middleware_benchmark/pull/18/commits/5fdfeaa208a88f57e1e9fe8ff25cc9e6bba47dd3 )

File: include/moveit_middleware_benchmark/scenarios/scenario_basic_service_client.hpp (hunk 5 above)
Comment (CihatAltiparmak) [reply include/moveit_middleware_benchmark/scenarios/scenario_basic_service_client.hpp]: Of course makes sensee. Solved in this commit ( https://github.com/CihatAltiparmak/moveit_middleware_benchmark/pull/18/commits/f46f58d1344319b1909ed2b4af87f2f74bb88556 )

**Comment by CihatAltiparmak** [comment]
Let's tag in a cool way. @sjahr Would you review again? I'm ready. Btw unfortunately industrial CI gives failure without any reason due to github action's pathetic abilities. So, i have created this PR to solve this problem. PR is here ( https://github.com/CihatAltiparmak/moveit_middleware_benchmark/pull/23 ).  

Additionaly to your review, i have used `Node` instead of `ExecuteProcess` here. Because ExecuteProcess bring about some problems in the termination of all processes. `add_two_ints` sometimes isn't terminated. For more, you can see the comment inside commit ( https://github.com/CihatAltiparmak/moveit_middleware_benchmark/pull/18/commits/ce7ad9f95e89d279a288e504b6adae1eda82047d ).

https://github.com/CihatAltiparmak/moveit_middleware_benchmark/blob/ce7ad9f95e89d279a288e504b6adae1eda82047d/launch/scenario_basic_service_client_benchmark.launch.py#L24-L28

**Review by henningkayser — APPROVED** [review]

**Pull request merged**